src/components/VmDetails/cards/DetailsCard.js

```diff
@@ -1,39 +1,258 @@
+      const actionResult = props.vm.getIn(['actionResults', state.correlationId])
+      if (actionResult) {
+        return { isEditing: false, isDirty: false, correlationId: null, correlatedMessages: null }
+      }
+      return {
+        correlatedMessages: props.userMessages.get('records').filter(
+          record => record.getIn([ 'failedAction', 'meta', 'correlationId' ]) === state.correlationId
+        ),
+      }
+    }
+
+    return null
+  }
+
+  componentDidUpdate (prevProps, prevState) {
+    if (prevState.isEditing && !this.state.isEditing) {
+      this.props.onEditChange(false)
+    }
+  }
+
+  handleCardOnStartEdit () {
+    this.setState({ isEditing: true, isDirty: false, correlationId: null, correlatedMessages: null })
+    this.props.onEditChange(true)
+  }
+
+  handleCardOnCancel () {
+    this.setState({ isEditing: false, isDirty: false, correlationId: null, correlatedMessages: null })
+    this.props.onEditChange(false)
+  }
+
+  handleCardOnSave () {
+    const { vm: stateVm } = this.state
+    const correlationId = generateUnique('DetailsCard-save_')
+
+    const vmUpdates = {
+      id: stateVm.get('id'),
+      //
+      // TODO: Select the proper fields to update
+      //
+    }
+    this.setState({ correlationId })
+    this.props.saveChanges(vmUpdates, correlationId)
+
+    return false
+  }
+
+  handleChange (fieldName, { target: { value } }) {
+    if (this.state.isEditing && !this.state.isDirty) {
+      this.props.onEditChange(true, true)
+    }
+
+    const newVm = this.state.vm.set(fieldName, value)
+    this.setState({ vm: newVm, isDirty: true })
+  }
+
+  render () {
+    const { vm, hosts, clusters, dataCenters, templates } = this.props
+    const { isEditing, correlatedMessages } = this.state
+
+    const hostName = (hosts && hosts.getIn([vm.get('hostId'), 'name'])) || '...'
+    const ip4Addresses = !vm.has('nics') ? [] : vm.get('nics').reduce((ipSet, nic) => [...ipSet, ...nic.get('ipv4')], [])
+    const fqdn = vm.get('fqdn') || '?'
+
+    const clusterName = (clusters && clusters.getIn([vm.getIn(['cluster', 'id']), 'name'])) || '...'
+
+    const dataCenterId = (clusters && clusters.getIn([vm.getIn(['cluster', 'id']), 'dataCenterId']))
+    const dataCenter = dataCenters && dataCenters.find(v => v.id === dataCenterId)
+    const dataCenterName = (dataCenter && dataCenter.name) || '...'
+
+    const templateName = (templates && templates.getIn([vm.getIn(['template', 'id']), 'name'])) || '...'
+
+    // look up in __storageDomains__ -- but, need to run 'fetchISOStorages' saga via 'getIsoStorageDomains' redux action to populate the store
+    const canChangeCD = isEditing && vm.get('status') === 'up'
+    const cdImageName = vm.getIn(['cdrom', 'fileId']) || msg.empty()
+    const isoList = createIsoList(this.props.storageDomains) // TODO: Try to make isoList part of state...
+    console.info('isoList', isoList)
```

Comment:
We have logger, so if you want to leave it, please, use logger.

Reply:
`import logger from '../logger'` uses the same log functions as `console.log` etc.  There is no difference.

Once I finish off the PR, the logging will be much further reduced.

Reply:
But there is a difference, logger add additional logic to `console.log` like checking debug mode. So it would be better to use `logger`. Or just remove this line.

Reply:
The logging will be mostly, if not fully, removed when I push next.

FWIW, console.(log|info|warn|error) are all proxied in logger.js such that importing the logger and using console.log actually call the same exact functions.  See #723.

Reply:
If `console.info` will be removed in this PR, then ok, but if not, then it must be rewrite to logger, because, no need to show this messages to user in production.